lib/flatpak-installation.c

```diff
@@ -1897,7 +1897,8 @@ flatpak_installation_update_full (FlatpakInstallation    *self,
   if (result == NULL)
     goto out;
 
-  if ((flags & FLATPAK_UPDATE_FLAGS_NO_DEPLOY) != 0)
```

Comment:
I would not be surprised if this was actually a bug in the upstream logic. Can you double check with @alexlarsson?
If we're doing a `NO_DEPLOY` update, it means we're only downloading new objects for an update, not undeploying the old version, and not deploying the new one.
So there should be nothing to prune in that case, as opposed to when we actually do a regular update or a `NO_PULL` one, as that will undeploy the old version which may result in objects being available for pruning...

Reply:
Yeah, agreed. We talked about this in London today, and we also concluded that the prune operations which flatpak does should be conditional on `!NO_PULL`, rather than anything to do with `NO_DEPLOY`. flatpak’s pruning is based on reachability from refs, and deploying/not deploying an app doesn’t affect its refs. Pulling/Not pulling an app does.

Reply:
This likely needs fixing in other places in flatpak too.

Reply:
I double checked with him and I believe he'd be fine with such a change. I'm working on it now